internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,29 @@
+package mapper
+
+import (
+	"fmt"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+)
+
+// Mapper defines the interface that Maps an incoming request to an aggregation key
+type Mapper interface {
```

Comment:
@jessicayuen continuing the [conversation](https://github.com/envoyproxy/xds-relay/pull/17#discussion_r391422717) here. I feel that the Mapper impl should closely match the proto and the spec defined in the doc. I don't foresee how Mapper can be overridden without diluting the spec from the design doc.

Reply:
makes sense, I think if we ever wanted to make this more customizable, it would be worth another spec / deeper discussion.

Reply:
makes sense.